backend/typescript/services/implementations/campService.ts

```diff
@@ -103,9 +106,47 @@ class CampService implements ICampService {
     }
   }
 
-  async getCampById(campId: string): Promise<GetCampDTO> {
-    let camp: Camp | null;
+  async getCampById(
+    campId: string,
+    campSessionId?: string,
+    waitlistedCamperId?: string,
+  ): Promise<GetCampDTO> {
+    if (waitlistedCamperId && campSessionId) {
+      try {
+        const waitlistedCamper: WaitlistedCamper | null = await MgWaitlistedCamper.findById(
+          waitlistedCamperId,
+        );
+
+        if (!waitlistedCamper || !waitlistedCamper?.linkExpiry) {
+          throw new Error(
+            `Waitlisted Camper with Id ${waitlistedCamperId} does not exist or does not have an invite link.`,
+          );
+        }
 
+        if (campSessionId !== waitlistedCamper.campSession.toString()) {
+          throw new Error(
+            `Given waitlisted camper is not on the waitlist for given camp session.`,
+          );
+        }
+
+        const linkExpiryDate = new Date(waitlistedCamper.linkExpiry);
+        const currentDate = new Date();
+        if (linkExpiryDate > currentDate) {
```

Comment:
should be <=? 
```suggestion
        if (linkExpiryDate <= currentDate) {
```

Reply:
I think if they are registering on the current date, it should be allowed, but good catch